### PR TITLE
fix: make sure backing array is fully populated when reslicing

### PR DESCRIPTION
**Pull request by kruskall**
solve nil pointer exception

### CODE REVIEW

**Review by marclop — APPROVED** [review]

File: input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go [hunk 1]
```diff
@@ -22,13 +22,23 @@ package modeldecoderutil
...
+
+			// populate the extra space
+			for ; idx < len(slice); idx++ {
+				slice[idx] = newFn()
 			}
 		}
-		slice = append([]model(slice)[:cap(slice)], extra...)
 	}
+
 	return slice[:n]
```
[the remark this answers is not in the document]
Comment (marclop) [reply input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go:43]: I don't follow. Could you elaborate?

[the remark this answers is not in the document]
Comment (marclop) [reply input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go:43]: The last reslice then it's only useful when we we're downsizing, correct?

[the remark this answers is not in the document]
Comment (marclop) [reply input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go:43]: well it's still really not obvious to my why we're doing what we're doing. Can you please document this thoroughly with a comment?

[the remark this answers is not in the document]
Comment (carsonip) [reply input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go:43]: Will it be safer to use [`slices.Grow`](https://pkg.go.dev/slices#Grow) and run `newFn` on the new elements instead?

[the remark this answers is not in the document]
Comment (carsonip) [reply input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go:43]: there's an exp package for older Go

**Pull request merged**